runtime/neumann/src/lib.rs

```diff
@@ -1156,6 +1157,34 @@ impl_runtime_apis! {
 		fn generate_task_id(account_id: AccountId, provided_id: Vec<u8>) -> Hash {
 			AutomationTime::generate_task_id(account_id, provided_id)
 		}
+
+		fn calculate_optimal_autostaking(
+			principal: i128,
+			collator: AccountId
+		) -> Result<AutostakingResult, Vec<u8>> {
+			let candidate_info = ParachainStaking::candidate_info(collator);
+			let money_supply = Balances::total_issuance() + Vesting::total_unvested_allocation();
+
+			let collator_stake = match candidate_info {
+				Some(x) => Ok(x.total_counted as i128),
+				None => Err("collator does not exist"),
+			}?;
+
+			let fee = (1 * DOLLAR) as i128;
+			let duration = 90;
+			// 24 = number of collators
```

Comment:
This is technically a dynamic value: `ParachainStaking::total_selected()`

Reply:
Discussed in Slack. Dynamic as in we configure it to a static number and not based on round. We definitely want to use this.